src/database.rs

```diff
@@ -31,41 +38,86 @@ pub enum SqlOption {
 }
 
 #[derive(Debug)]
-pub struct DbError {
-    description: String,
-    cause: Option<String>,
+pub enum DbError {
+    Error(String),
+    PoolError(r2d2::InitializationError),
+    PlatformError(PlatformError),
 }
 
-/// rough implementation of Database errors
-impl DbError{
+impl DbError {
     pub fn new(description: &str) -> Self {
-        DbError {
-            description: description.to_string(),
-            cause: None,
+        DbError::Error(description.to_string())
+    }
+}
+
+impl Error for DbError {
+    fn description(&self) -> &str {
+        match *self {
+            DbError::Error(ref description) => description,
+            DbError::PoolError(ref err) => err.description(),
+            DbError::PlatformError(ref err) => err.description(),
         }
     }
 
-    pub fn from_string(description: String) -> Self {
-        DbError {
-            description: description,
-            cause: None,
+    fn cause(&self) -> Option<&Error> {
+        match *self {
+            DbError::Error(_) => None,
+            DbError::PoolError(ref err) => Some(err),
+            DbError::PlatformError(ref err) => Some(err),
         }
     }
 }
 
-impl Error for DbError{
-    fn description(&self) -> &str {
-        &self.description
+impl fmt::Display for DbError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            DbError::PoolError(ref err) => write!(f, "Pool error: {}", err),
+            DbError::PlatformError(ref err) => write!(f, "PostgreSQL error: {}", err),
+            DbError::Error(_) => write!(f, "{}", self.description()),
+        }
     }
+}
 
-    fn cause(&self) -> Option<&Error> {
-        None
+impl From<r2d2::InitializationError> for DbError {
+    fn from(err: r2d2::InitializationError) -> Self {
+        DbError::PoolError(err)
     }
 }
 
-impl fmt::Display for DbError {
-    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
-        write!(fmt, "{}", self.description())
+impl From<PlatformError> for DbError {
+    fn from(err: PlatformError) -> Self {
+        DbError::PlatformError(err)
+    }
+}
+
+impl From<RegexError> for DbError {
+    fn from(err: RegexError) -> Self {
+        DbError::new(err.description())
+    }
+}
+
+impl From<PgError> for DbError {
+    fn from(err: PgError) -> Self {
+        DbError::PlatformError(From::from(err))
+    }
+}
+
+impl From<PgConnectError> for DbError {
+    fn from(err: PgConnectError) -> Self {
+        DbError::PlatformError(From::from(err))
+    }
+}
+
+impl From<MyError> for DbError {
+    fn from(err: MyError) -> Self {
+        DbError::PlatformError(From::from(err))
+    }
+}
+
+#[cfg(feature = "sqlite")]
+impl From<SqliteError> for DbError {
```

Comment:
There seem to be a compile error here https://github.com/ivanceras/rustorm/issues/19

The error the trait `core::convert::From<rusqlite::SqliteError>` is not implemented for the type `database::DbError`
but the code here states otherwise.